version: '3.4'

services:
  autonomous_ros2:
    image: autonomous
    runtime: nvidia
    build:
      context: ..
      dockerfile: .devcontainer/ros2_ws.Dockerfile

    environment:
      - DISPLAY
      - NVIDIA_VISIBLE_DEVICES=0
      - NVIDIA_DRIVER_CAPABILITIES=all
      
      - ROS_DOMAIN_ID=47  # Its 47 for obvious reasons

    volumes:
      - ..:/ws
      - /dev:/dev

    network_mode: host
    ipc: host
    stdin_open: true
    tty: true

    deploy:
      resources:
        reservations:
          devices:
            - driver: nvidia
              device_ids: []
              capabilities: [gpu]
    devices:
      - /dev/ttyUSB0:/dev/ttyUSB0


  autonomous_zed:
    image: zed_sdk
    runtime: nvidia
    privileged: true
    
    build:
      context: ..
      dockerfile: .devcontainer/zed.Dockerfile

    environment:
      - DISPLAY
      - NVIDIA_VISIBLE_DEVICES=0
      - NVIDIA_DRIVER_CAPABILITIES=all
      - ROS_DOMAIN_ID=47

    # command: bash ". ~/.bashrc && ros2 launch zed_wrapper zed_camera.launch.py camera_model:='zed2i'"

    volumes:
      - ../src/perception/:/ws/src/perception/
      - ../src/moa/moa_description:/ws/src/moa/moa_description
      - ../src/moa/moa_msgs:/ws/src/moa/moa_msgs
      

    network_mode: host
    ipc: host
    stdin_open: true
    tty: true

    deploy:
      resources:
        reservations:
          devices:
            - driver: nvidia
              device_ids: []
              capabilities: [gpu]
